src/natives/resize/win/Program.cs

```diff
@@ -0,0 +1,137 @@
+        public struct Rect {
+            public int left;
+            public int top;
+            public int right;
+            public int bottom;
+        }
+
+        [DllImport("user32.dll")]
+        private static extern IntPtr GetForegroundWindow();
+
+        [DllImport("user32.dll")]
+        static extern uint GetWindowThreadProcessId(IntPtr hWnd, IntPtr ProcessId);
+
+        [DllImport("kernel32.dll")]
+        static extern uint GetCurrentThreadId();
+
+        [DllImport("user32.dll")]
+        static extern bool AttachThreadInput(uint idAttach, uint idAttachTo,
+           bool fAttach);
+
+        [DllImport("user32.dll", SetLastError = true)]
+        static extern bool BringWindowToTop(IntPtr hWnd);
+
+        [DllImport("user32.dll")]
+        static extern bool IsZoomed(IntPtr hWnd);
+
+        [DllImport("user32.dll")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        static extern bool SetForegroundWindow(IntPtr hWnd);
+
+        [DllImport("user32.dll")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        static extern bool ShowWindow(IntPtr hWnd, int nCmdShow);
+
+        static void RestoreDownWindow (IntPtr hWnd){
+            IntPtr savedForegroundWindow = GetForegroundWindow();
+
+            ShowWindow(hWnd, SW_RESTORE);
+
+            if (GetForegroundWindow() != savedForegroundWindow)
+                ForceForegroundWindow(savedForegroundWindow);
+        }
+
+        static void ResizeWindow(IntPtr hWnd, int width, int height)
+        {
+            WindowInfo wi = new WindowInfo();
+            GetWindowInfo(hWnd, ref wi);
+
+            MoveWindow(hWnd, wi.rcWindow.left, wi.rcWindow.top, width, height, true);
+        }
+
+        private static void ForceForegroundWindow(IntPtr hWnd)
+        {
+            uint foreThread = GetWindowThreadProcessId(GetForegroundWindow(), IntPtr.Zero);
+            uint appThread = GetCurrentThreadId();
+            bool threadAccessRequired = foreThread != appThread;
+
+            if(threadAccessRequired)
+                AttachThreadInput(foreThread, appThread, true);
+
+            BringWindowToTop(hWnd);
+
+            ShowWindow(hWnd, IsZoomed(hWnd) ? SW_SHOWMAXIMIZED : SW_SHOW);
+
+            if(threadAccessRequired)
+                AttachThreadInput(foreThread, appThread, false);
+        }
+
+        static void Main(string[] args) {
+            if(args.Length != 3) {
+                Console.Error.Write("Incorrect arguments");
+                Environment.Exit(-1);
+            }
+
+            IntPtr hWnd = (IntPtr)int.Parse(args[0]);
+
+            int width  = Convert.ToInt32(args[1]),
+                height = Convert.ToInt32(args[2]);
+
+            //NOTE: if the target window is maximized we should restore it before resizing.
+            // Otherwise WS_MAXIMIZED window style will still be applied to the window, that results in wrong icon in header
+            // and buggy behaviour in some cases.
```

Comment:
NOTE: If the target window is maximized, we should restore it before resizing. Otherwise, the WS_MAXIMIZED style will still be applied to the window, which will result in a wrong icon displayed in the header and buggy behavior in certain cases.